lib/chef/provider/group/solaris.rb

```diff
@@ -0,0 +1,72 @@
+#
+# Author:: Joshua Justice (<jjustice6@bloomberg.net>)
+# Copyright:: Copyright 2008-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/group/groupadd"
+
+class Chef
+  class Provider
+    class Group
+      class Solaris < Chef::Provider::Group::Groupadd
+
+        provides :group, platform: "solaris2"
```

Comment:
```suggestion
        # this provides line is setup to only catch the solaris2 platform and not other platforms in the Solaris platform_family      
        provides :group, platform: "solaris2"
```

Reply:
Tweaked comment further and did a separate commit so i could sign-off-by without having to do a force push like the others.